site/_posts/2017-10-24-api-contributors-guide.md

```diff
@@ -0,0 +1,250 @@
+---
+title: API Contributor's Guide
+author: jntullo
+date: 2017-10-24
+tags: collaboration developers tutorials api
+comments: true
+published: true
+---
+
+The API has been growing quickly thanks to our many contributors. In an effort to help new contributors get up to speed quickly, it was about time for a blog post to explain the process of adding in a new collection and subcollection, as well as provide some examples for commonly asked questions.
+
+The [api.yml](https://github.com/ManageIQ/manageiq-api/blob/master/config/api.yml) is where a lot of the magic :sparkles: happens. The best way to explain it is through an example, so let's walk through adding in a new collection and subcollection.
+
+We'll keep the example generic through the creation of a `Coffee` API. Because who doesn't need more :coffee: in their life?
+
+## Defining a new collection with basic CRUD
+The new `Coffee` collection will need to be defined under [collections](https://github.com/ManageIQ/manageiq-api/blob/master/config/api.yml#L37) in the `api.yml`. Note that all of the collections are in alphabetical order. 
+The collection will need to return details about the coffee, update them if they need some more sugar, create new ones (Pumpkin, anyone?), and delete any we're not a fan of - basic CRUD. 
+
+Here is a glimpse into what adding the new collection with basic CRUD will look like:
+```
+:collections:
+    :coffee:
+        :description: Coffee
+        :options: 
+        - :collection
+        :verbs: *gpd
+        :klass: Coffee
+        :collection_actions:
+            :get: 
+            - :name: read
+              :identifier: coffee_show_list 
+            :post:
+            - :name: create
+              :identifier: coffee_new 
+            - :name: edit
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+        :resource_actions:
+            :get:
+            - :name: read
+              :identifier: coffee_show 
+            :post:
+            - :name: edit
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+            :delete:
+            - :name: delete
+              :identifier: coffee_delete
+```
+
+
+Here is a breakdown of what some of these elements mean:
+* **: coffee:**
+    * Defines the route /api/coffee
```

Comment:
So...to be super technical 😀 it defines the name in that portion of the URI....it's the `:collection`/`:subcollection` option in conjunction with this name and supported verbs/actions that defines the routes (:weary:)